test/test_utilities.jl

```diff
@@ -171,6 +171,13 @@ function test_show()
     @test curvature(p) == Convex.ConvexVexity()
 
     @test sprint(show, p) == """
+    summary
+    ├─ # variables    : 1 (1 scalar elements)
+    ├─ # constraints  : 2 (2 scalar elements)
+    ├─ # coefficients : 2
+    ├─ # atoms        : 3
+    └─ size           : 624 bytes
```

Comment:
I guess my question is: what information is useful here?

It's handy to know if:
 * There are lots of variables, both matrix and scalar valued
 * There are lots of constraints, both matrix and scalar valued
 * If there are lots of data matrices (`# coefficients` is the length of `Matrix` including zeros plus non-zeros in a sparse matrix)
 * There are lots of atoms (big expressions tree)
 * The problem uses a lot of memory

Things like what cones it lowers to etc are a bit more abstract. We should probably add default printing to MOI.